patch-db/src/locker/bookkeeper.rs

```diff
@@ -35,11 +36,13 @@ impl LockBookkeeper {
     pub fn lease(
         &mut self,
         req: Request,
-    ) -> Result<Option<oneshot::Receiver<LockInfo>>, LockError> {
+    ) -> Result<Option<oneshot::Receiver<LockInfos>>, LockError> {
         #[cfg(feature = "unstable")]
-        if let Err(e) = self.order_enforcer.try_insert(&req.lock_info) {
-            req.reject(e.clone());
-            return Err(e);
+        for info in req.lock_info.as_vec() {
+            if let Err(e) = self.order_enforcer.try_insert(info) {
+                req.reject(e.clone());
+                return Err(e);
+            }
```

Comment:
this will not work. We have to consider all order enforcement in isolation, or we have to guarantee that the vec is sufficiently ordered that the batch doesn't conflict with itself.